uint32 button_index
float64 command_code